src/App.tsx

```diff
@@ -15,47 +16,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+  RESET,
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortBy] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  let visibleGoods = [...goodsFromServer];
+
+  switch (sortType) {
+    case SortType.NONE:
+      break;
+
+    case SortType.ALPHABET:
+      visibleGoods = visibleGoods.sort(
+        (a, b) => (a.localeCompare(b.toString())
```

Comment:
Missing closing parenthesis after `a.localeCompare(b` in the sort function, should be `a.localeCompare(b)`